src/qumada/instrument/mapping/Harvard/Decadac.py

```diff
@@ -114,36 +110,35 @@ def setup_trigger_in(self, trigger_settings: dict):
             "tracking_pulse": 7,
             "digital": 6,
         }
-        print("Warning: The Decadacs trigger level is fixed at roughly 1.69 V and cannot be changed. \
-              Please make sure that your triggers are setup accordingly")
+        print(
+            "Warning: The Decadacs trigger level is fixed at roughly 1.69 V and cannot be changed.\nPlease make sure that your triggers are setup accordingly."
+        )
         trigger_mode = trigger_settings.get("trigger_mode", "continuous")
         polarity = trigger_settings.get("trigger_mode_polarity", "positive")
-        
-        match (trigger_mode, polarity):
-            case ("edge", "positive"): 
-                mode = 12
-            case ("edge", "negative"):
-                mode = 14
-            case ("digital", "positive"):
-                mode = 10
-            case("digital", "negative"):
-                mode = 2
-            # TODO: CHeck other cases
-            case("continuous", "positive"):
-                mode = 0
-            case("continuous", "negative"):
-                mode = 0
-            case _:
-                raise Exception("Selected trigger mode is not supported by DecaDac")
-                
+
+        if (trigger_mode, polarity) == ("edge", "positive"):
+            mode = 12
+        elif (trigger_mode, polarity) == ("edge", "negative"):
+            mode = 14
+        elif (trigger_mode, polarity) == ("digital", "positive"):
+            mode = 10
+        elif (trigger_mode, polarity) == ("digital", "negative"):
+            mode = 2
+        elif (trigger_mode, polarity) == ("continuous", "positive"):
+            mode = 0
+        elif (trigger_mode, polarity) == ("continuous", "negative"):
+            mode = 0
+        else:
+            raise Exception("Selected trigger mode is not supported by DecaDac")
```

Comment:
This code was not compatible with python 3.9. Refactored in PR #11.

Reply:
Is there any reason to maintain downward compatibility to 3.9?

Reply:
It's probably not a huge deal to switch to 3.10 as minimum python version. Although I like that we keep compatibility with QCoDeS. Switching just for the structural pattern matching is probably not worth losing that.